kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -542,6 +659,216 @@ namespace Kratos
+        Vector &rElementalDistancesExtraVector)
+	{
+		constexpr std::size_t n_nodes = TDim + 1;
+		constexpr std::size_t n_edges = (TDim == 2) ? 3 : 6;
+		const auto &r_geometry = rElement.GetGeometry();
+
+		// Initialize the elemental distances with extrapolated intersections to the domain characteristic length
+		const double initial_distance = CalculateCharacteristicLength();
+		if (rElementalDistancesExtraVector.size() != n_nodes) {
+			rElementalDistancesExtraVector.resize(n_nodes, false);
+		}
+		for (std::size_t i_node = 0; i_node < n_nodes; ++i_node) {
+			rElementalDistancesExtraVector[i_node] = initial_distance;
+		}
+
+		// Get combined vector of intersection and extrapolated edge ratios and count total amount of cut edges
+		array_1d<double, (TDim == 2) ? 3 : 6> combined_edge_ratios = array_1d<double, n_edges>(n_edges, -1.0);
+		std::size_t n_cut_edges = 0;
+		for (std::size_t i_edge = 0; i_edge < n_edges; i_edge++) {
+			if (rCutEdgesRatioVector[i_edge] > 0.0){
+				combined_edge_ratios[i_edge] = rCutEdgesRatioVector[i_edge];
+				n_cut_edges++;
+			} else if (rCutExtraEdgesRatioVector[i_edge] > 0.0) {
+				combined_edge_ratios[i_edge] = rCutExtraEdgesRatioVector[i_edge];
+				n_cut_edges++;
+			} else {
+				combined_edge_ratios[i_edge] = -1.0;
+			}
+		}
+
+		// Check if element qualifies as being intersected after calculating extrapolated intersections and calculate elemental distances
+		if (n_cut_edges >= TDim) {
+			// Calculate points from nodes of edges and length ratio of intersections
+			std::vector<array_1d <double,3> > intsect_pts_vector;
+			ConvertRatiosToIntersectionPoints(r_geometry, combined_edge_ratios, intsect_pts_vector);
+
+			// If there are more than 3 intersected edges (3D), compute the least squares plane approximation
+			// by using the ComputePlaneApproximation utility. Otherwise, the distance is computed using
+			// the plane defined by the 3 intersection points (3D).
+			const bool do_plane_approx = (n_cut_edges == TDim) ? false : true;
+
+			if (do_plane_approx){
+				// Call the plane optimization utility
+				array_1d<double,3> base_pt, normal;
+				ComputePlaneApproximation(rElement, intsect_pts_vector, base_pt, normal);
+
+				// Compute the distance to the approximation plane
+				Plane3D approximation_plane(normal, Point{base_pt});
+				for (uint8_t i = 0; i < n_nodes; i++) {
+					rElementalDistancesExtraVector[i] = approximation_plane.CalculateSignedDistance(r_geometry[i]);
+				}
+			} else {
+				// Create a plane with the 3 intersection points (or 2 in 2D)
+				Plane3D plane = SetIntersectionPlane(intsect_pts_vector);
+
+				// Compute the distance to the intersection plane
+				for (uint8_t i = 0; i < n_nodes; i++) {
+					rElementalDistancesExtraVector[i] = plane.CalculateSignedDistance(r_geometry[i]);
+				}
+			}
+
+			// Correct the distance values orientation
+			CorrectDistanceOrientation(r_geometry, rIntersectedObjects, rElementalDistancesExtraVector);
```

Comment:
This part of the code seems to be copied and pasted from the standard elemental distance calculation? Couldn't we reuse that function in here?

Reply:
I thought about that. It would mean changing that function and putting the main part of it into another function, that we could call then in here.

Reply:
It's quite similar, but not the same, it would result in the same amount of functions I think.

Reply:
Exactly. I'd do something like this. Implement the common part in an auxiliary `private` function.